client/me/purchases/remove-purchase/index.jsx

```diff
@@ -45,12 +80,41 @@ const RemovePurchase = React.createClass( {
 		this.setState( { isDialogVisible: true } );
 	},
 
+	changeSurveyStep() {
+		this.setState( {
+			surveyStep: this.state.surveyStep === 1 ? 2 : 1,
+		} );
+	},
+
 	removePurchase( closeDialog ) {
 		this.setState( { isRemoving: true } );
 
 		const purchase = getPurchase( this.props ),
 			{ selectedSite, user } = this.props;
 
+		if ( ! isDomainRegistration( purchase ) && config.isEnabled( 'upgrades/removal-survey' ) ) {
+			const survey = wpcom.marketing().survey( 'calypso-remove-purchase', this.props.selectedSite.ID );
+			survey.addResponses( {
+				'why-cancel': {
+					response: this.state.questionOneRadio,
+					text: this.state.questionOneText
+				},
+				'next-adventure': {
+					response: this.state.questionTwoRadio,
+					text: this.state.questionTwoText
+				},
+				'what-better': { text: this.state.questionThreeText }
+			} );
+
```

Comment:
before you submit the survey, you may find it helpful to emit a debug line, so you can more easily test this in the wild.